multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/template/TemplateFactory.java

```diff
@@ -132,6 +280,44 @@ private void fillTemplateSchema(final ClassTemplate classTemplate, final Boolean
         rootSchema.put("packageModel", classTemplate.getModelPackage());
       }
       fillTemplate(filePath.toString(), schemaObject.getClassName(), templateName, rootSchema);
+      for (SchemaFieldObject fieldObject : schemaObject.getFieldObjectList()) {
+        if (fieldObject.isRequired()) {
+          fillTemplateNotNull(filePath, classTemplate.getModelPackage());
+          fillTemplateNotNullValidator(filePath, classTemplate.getModelPackage());
+        }
+        if (Objects.nonNull(fieldObject.getMaximum())) {
+          fillTemplateMax(filePath, classTemplate.getModelPackage());
+          fillTemplateMaxValidator(filePath, classTemplate.getModelPackage(), schemaObject);
+        }
+        if (Objects.nonNull(fieldObject.getMaxItems())) {
+          fillTemplateMaxItems(filePath, classTemplate.getModelPackage());
+          fillTemplateMaxItemsValidator(filePath, classTemplate.getModelPackage());
+        }
+        if (Objects.nonNull(fieldObject.getMinimum())) {
+          fillTemplateMin(filePath, classTemplate.getModelPackage());
+          fillTemplateMinValidator(filePath, classTemplate.getModelPackage(), schemaObject);
+        }
+        if (Objects.nonNull(fieldObject.getMinItems())) {
+          fillTemplateMinItems(filePath, classTemplate.getModelPackage());
+          fillTemplateMinItemsValidator(filePath, classTemplate.getModelPackage());
+        }
+        if (Objects.nonNull(fieldObject.getMinLength()) || Objects.nonNull(fieldObject.getMaxLength())) {
+          fillTemplateSize(filePath, classTemplate.getModelPackage());
+          fillTemplateSizeValidator(filePath, classTemplate.getModelPackage());
+        }
+        if (Objects.nonNull(fieldObject.getPattern())){
+          fillTemplatePattern(filePath, classTemplate.getModelPackage());
+          fillTemplatePatternValidator(filePath, classTemplate.getModelPackage());
+        }
+        if (Objects.nonNull(fieldObject.getMultipleOf())){
+          fillTemplateMultipleOf(filePath, classTemplate.getModelPackage());
+          fillTemplateMultipleOfValidator(filePath, classTemplate.getModelPackage(), schemaObject);
+        }
+        if (Objects.nonNull(fieldObject.getUniqueItems())){
+          fillTemplateUniqueItems(filePath, classTemplate.getModelPackage());
+          fillTemplateUniqueItemsValidator(filePath, classTemplate.getModelPackage());
+        }
+      }
```

Comment:
Running this for every field seems wasteful. We should generate these only once. Also, the checks themselves can probably be moved elsewhere to improve readability.

Reply:
Done